src/components/Card/Card.jsx

```diff
@@ -7,36 +9,45 @@ import {
   ShowInfo,
   CardFunction,
 } from './Card';
-import likeSrc from '../../assets/heart.png';
-import { useNavigate } from 'react-router-dom';
-import { useState } from 'react';
-import Button from '../../components/Button/Button.jsx';
+import Button from '../Button/Button.jsx';
 
 const Card = ({ show }) => {
-  const { imageUrl, showId, title, theaterNames, startDate, endDate } = show;
-
   const navigate = useNavigate();
   const [isClicked, setIsClicked] = useState(false);
   const token = localStorage.getItem('token');
+  const { wishId, imageUrl, showId, title, theaterNames, startDate, endDate } =
+    show;
 
-  function handleLike() {
+  function handleLike(id) {
     setIsClicked(prev => !prev);
-    !isClicked &&
-      fetch('http://10.58.52.53:8080/wishs/create', {
-        method: 'POST',
-        headers: {
-          'Content-Type': 'application/json;charset=utf-8',
-          Authorization: token,
-        },
-        body: JSON.stringify({ showId: 4 }),
-      }).then(res => res.json());
+    isClicked
+      ? fetch(`${APIS.wish}?wishId=${id}`, {
+          method: 'DELETE',
+          headers: {
+            'Content-Type': 'application/json;charset=utf-8',
+            Authorization: token,
+          },
+        }).then(res => {
+          if (res.status === 204) {
+            return;
+          }
+        })
+      : fetch(`${APIS.wish}/create`, {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json;charset=utf-8',
+            Authorization: token,
+          },
+          body: JSON.stringify({ showId: `${showId}` }),
```

Comment:
body에 showId가 string 형태로 담겨야 하는게 아니라면 
```suggestion
          body: JSON.stringify({ showId: showId }),
```
또는
```suggestion
          body: JSON.stringify({ showId }),
```
로 축약해서 작성할 수 있겠습니다!